incubator/redis-auto-cluster/templates/config.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "redis-auto-cluster.name" . }}
+    chart: {{ template "redis-auto-cluster.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  redis.conf: |+
+    cluster-enabled yes
+    cluster-require-full-coverage no
+    cluster-node-timeout {{ .Values.config.clusterNodeTimeout }}
+    cluster-config-file /data/nodes.conf
+    cluster-migration-barrier 1
+
+    protected-mode no
+
+    maxmemory {{ .Values.shards.sizeGb }}Gb
+    maxmemory-policy {{ .Values.config.maxMemoryPolicy }}
+
+    appendonly {{ .Values.config.appendOnly }}
+
+{{ .Values.config.saveOptions | indent 4 }}
```

Comment:
You could specify anything here, not just save options. Why not make the whole config configurable?

Reply:
That's not that simple, because in order to get the cluster properly configured some of the settings must be clearly defined. Allowing the whole config to be provided opens the door to wrong settings or required ones missing. As you see I compute the max memory  so that it matches the pod resources as well. I'm not sure what to do to be honest. 
I have 2 clusters built with this chart running in production and staging since 1+ month and no issues since.